### PR TITLE
Changed cluster client api to use IEnumerable for InitialContacts.

### PR DESCRIPTION
**Pull request by Danthar**
And updated usage patterns in tests and examples

### CODE REVIEW

**Comment by Danthar** [comment]
Will need probably api approval. But i'm going to wait after review.

**Comment by Aaronontheweb** [comment]
`ClusterClientConfigSpec.ClusterClientSettings_must_throw_exception_on_empty_initial_contacts` - failed.

File: src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClientSettings.cs [hunk 1]
```diff
@@ -113,7 +113,7 @@ public static ClusterClientSettings Create(Config config)
         /// <param name="reconnectTimeout">TBD</param>
         /// <exception cref="ArgumentException">TBD</exception>
         public ClusterClientSettings(
-            IImmutableSet<ActorPath> initialContacts,
+            IEnumerable<ActorPath> initialContacts,
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools/Client/ClusterClientSettings.cs]: Looks good to me.

File: src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/ClusterClient/ClusterClientStartSpecConfig.cs [hunk 2]
```diff
@@ -69,15 +70,13 @@ protected ClusterClientStartSpec(ClusterClientStartSpecConfig config) : base(con
 
             protected override int InitialParticipantsValueFactory => 3;
 
-            private ImmutableHashSet<ActorPath> InitialContacts
+            private IEnumerable<ActorPath> InitialContacts
             {
                 get
                 {
                     //return new List<ActorPath>().ToImmutableHashSet();
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/ClusterClient/ClusterClientStartSpecConfig.cs]: Delete this.

**Comment by alexvaluyskiy** [comment]
@Danthar We can't merge this Pull Request anymore, because Cluster Tools become stable. Just make another constructor and overloaded method `WithInitialContacts`

**Comment by Danthar** [comment]
I was going to update this PR. But I changed my mind. Adding a constructor overload defeats the purpose of this PR.

**Pull request closed without merging**